src/Persistence.LinnApps/Repositories/GoodsInLogEntryRepository.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace Linn.Stores2.Persistence.LinnApps.Repositories
+{
+    using System;
+    using System.Linq;
+    using System.Linq.Expressions;
+
+    using Linn.Common.Persistence.EntityFramework;
+    using Linn.Stores2.Domain.LinnApps.GoodsIn;
+
+    public class GoodsInLogEntryRepository : EntityFrameworkRepository<GoodsInLogEntry, int>
+    {
+        private readonly ServiceDbContext serviceDbContext;
+
+        public GoodsInLogEntryRepository(ServiceDbContext serviceDbContext) : base(serviceDbContext.GoodsInLogEntries)
+        {
+            this.serviceDbContext = serviceDbContext;
+        }
+
+        public override IQueryable<GoodsInLogEntry> FindAll()
+        {
+            return this.serviceDbContext.GoodsInLogEntries
+                .Where(g => g.DateCreated >= DateTime.Now.AddDays(-14))
+                .OrderByDescending(g => g.DateCreated);
+        }
+
+        public override IQueryable<GoodsInLogEntry> FilterBy(Expression<Func<GoodsInLogEntry, bool>> expression)
+        {
+            return this.serviceDbContext.GoodsInLogEntries.Where(expression).OrderByDescending(g => g.DateCreated);
```

Comment:
I would order by the id as opposed to the date probably - might make the query run faster

Reply:
+ if you go down the reporting route you can delete this repository implementation and just use the generic one since you can achieve any ordering you need in your report service